src/librustc_codegen_llvm/coverageinfo/mod.rs

```diff
@@ -97,8 +97,11 @@ impl CoverageInfoBuilderMethods<'tcx> for Builder<'a, 'll, 'tcx> {
     }
 }
 
-pub(crate) fn write_filenames_section_to_buffer(filenames: &Vec<CString>, buffer: &RustString) {
-    let c_str_vec = filenames.iter().map(|cstring| cstring.as_ptr()).collect::<Vec<_>>();
+pub(crate) fn write_filenames_section_to_buffer<'a>(
+    filenames: impl IntoIterator<Item = &'a CString>,
+    buffer: &RustString,
+) {
+    let c_str_vec = filenames.into_iter().map(|cstring| cstring.as_ptr()).collect::<Vec<_>>();
```

Comment:
Just out of curiosity, is this arguably not a bit of an antipattern, as elaborated in [this recent comment on Reddit](https://www.reddit.com/r/rust/comments/i44ahg/google_engineers_just_submitted_a_new_llvm/g0ghpi6/)? (Here and in other commits as well.) I.e., we'd maybe want to leave the signature as is and change the callsites instead.

Perhaps in the case of the changes in this PR, there isn't actually any more monomorphization taking place as a result, but I was just curious about this possibility since we do see a perf hit.

Reply:
This crate-local and there's really only one caller, but it just didn't seem useful to expose the fact that it's getting a `&IndexSet` at this level. The function only needs _something_ that provides `&CString`, but I think it shouldn't really matter what. I don't want this function to continue taking `&Vec<CString>` though, because that would require a new collection at the caller side.

> (Here and in other commits as well.)

Can you cite more examples? If I increased genericity badly, I didn't realize it. I know there are a few new methods where I return `impl Iterator`, but that's a single existential type.

Reply:
Anyway, I think that antipattern is more about having a generic intro in the function body, followed by a lot of code that's not actually generic. In this case, the following code is just a single FFI call, which isn't interesting to deduplicate even if there were a lot of different monomorphizations here.

Reply:
Fair enough! Re: "in other commits as well", I appear to have been mistaken, sorry. Upon looking again I don't really see anything that could be an issue.

Reply:
No problem, thanks for looking!